app/src/lib/stores/app-store.ts

```diff
@@ -5541,12 +5539,32 @@ export class AppStore extends TypedBaseStore<IAppState> {
     headCloneURL: string,
     headRefName: string
   ): Promise<Branch | null> {
-    const gitHubRepository = repository.gitHubRepository
-    const isRefInThisRepo = headCloneURL === gitHubRepository.cloneURL
-    const isRefInUpstream =
-      gitHubRepository.parent !== null &&
-      headCloneURL === gitHubRepository.parent.cloneURL
+    const { cloneURL, parent } = repository.gitHubRepository
+    const gitStore = this.gitStoreCache.get(repository)
+
+    let remote = null
+
+    // Determine whether the ref is in the current repository or the parent
+    if (headCloneURL === cloneURL) {
+      remote = gitStore.defaultRemote
+    } else if (headCloneURL === parent?.cloneURL) {
+      remote = gitStore.upstreamRemote
+    }
+
+    // If we don't have a remote here, it's probably going
+    // to just crash and burn on checkout, but that's okay
```

Comment:
Maybe I'm missing something, but if I understand the logic correctly having a null `remote` here is a possible scenario: it'll just trigger the logic in L5504 where it tries to find the fork remote repo or add it to the current repo.

In that case, we can get rid of this comment:

```suggestion
```

Reply:
Yeah, this comment came from the earlier logic and was specifically related to not having a _default_ remote. I'm 👍 on removing it.